packages/animations/browser/src/dsl/animation_timeline_builder.ts

```diff
@@ -897,17 +880,18 @@ function roundOffset(offset: number, decimalPoints = 3): number {
   return Math.round(offset * mult) / mult;
 }
 
-function flattenStyles(input: (ɵStyleData|string)[], allStyles: ɵStyleData) {
-  const styles: ɵStyleData = {};
+function flattenStyles(
+    input: Array<(Map<string, string|number>| string)>, allStyles: Map<string, string|number>) {
+  const styles = new Map<string, string|number>();
   let allProperties: string[];
   input.forEach(token => {
     if (token === '*') {
-      allProperties = allProperties || Object.keys(allStyles);
+      allProperties = allProperties || [...allStyles.keys()];
```

Comment:
Looks like we may avoid creating an extra array here?
```suggestion
      allProperties = allProperties || allStyles.keys();
      for (const prop of allProperties) { ... }
```

Reply:
Sounds good.